src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -1,12 +1,13 @@
 @using Microsoft.AspNetCore.Builder
 @using Microsoft.Extensions.Options
+@using Microsoft.AspNetCore.Http.Extensions
 @inject Microsoft.AspNetCore.Mvc.Localization.IViewLocalizer T
 @inject OrchardCore.IOrchardHelper Orchard
 @inject IOptions<RequestLocalizationOptions> LocOptions
 @{
-    var fullUri = Context.Request.Scheme + "://" + Context.Request.Host.ToUriComponent();
-    fullUri += Context.Request.PathBase.HasValue ? Context.Request.PathBase.ToUriComponent() : "";
-    fullUri += Context.Request.Path.ToUriComponent();
+    Uri currentUrl = new Uri(Context.Request.GetEncodedUrl());
+    var urlWithTenant = String.Format("{0}{1}{2}{3}", Context.Request.Scheme,
+                    Uri.SchemeDelimiter, currentUrl.Authority, currentUrl.AbsolutePath);
```

Comment:
@Skrypt 

I think you just need to set an uri string with `var uri = Context.Request.GetEncodedUrl()`

It already takes into account the `Scheme` and the `Host` ans so on, [see here](https://github.com/dotnet/aspnetcore/blob/master/src/Http/Http.Extensions/src/UriHelper.cs#L177)

Reply:
Yes but the javascript changes the ui-culture=culturecode part of the url and Context.Request.GetEncodedUrl() returns the entire thing. So I just need to remove the querystring part so that it gets appended through javascript.